search-app/src/cmr/search/services/acls/collections_cache.clj

```diff
@@ -16,16 +23,39 @@
 (defn create-cache
   "Creates a new empty collections cache."
   []
-  (mem-cache/create-in-memory-cache))
+  ;;(mem-cache/create-in-memory-cache)
+  (redis-cache/create-redis-cache {:keys-to-track [:collections-for-gran-acls] :ttl (* 15 60)}))
 
-(def cache-key
-  :collections-for-gran-acls)
+(defn- clj-times->time-strs
+  "Take a map and convert any date objects into strings so the map can be cached.
+   This can be reversed with time-strs->clj-times."
+  [data]
+  (walk/postwalk
+   #(if (true? (instance? org.joda.time.DateTime %))
+      (cmr.common.date-time-parser/clj-time->date-time-str %)
+      %)
+   data))
+
+(defn- time-strs->clj-times
+  "Take a map which has string dates and convert them to DateTime objects. This
+   can be reversed with clj-times->time-strs."
+  [data]
+  (walk/postwalk
+   #(if (and (instance? String %) (cmr.common.date-time-parser/try-parse-datetime %))
+      (cmr.common.date-time-parser/parse-datetime %)
+      %)
+   data))
 
 (defn- fetch-collections
   "Executes a query that will fetch all of the collection information needed for caching."
   [context]
+  ;; when creating a result processor, realize all the lazy (delay) values to
+  ;; actual values so that the resulting object can be cached in redis or some
+  ;; other text based caching system and not clojure memory
   (let [result-processor (fn [_ _ elastic-item]
-                           (assoc (acl-rhh/parse-elastic-item :collection elastic-item)
+                           (assoc (util/delazy-all (acl-rhh/parse-elastic-item
```

Comment:
can you use a (doall (acl-rhh/parse-elastic-item :collection elastic-item))?

Reply:
need to come back to this, at first inspection I think that lazy is not the same as delay. I have some code to show this, but lets talk about it and see if I'm missing something.